MedallionShell/AttachedCommand.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Medallion.Shell.Streams;
+
+namespace Medallion.Shell
+{
+    internal sealed class AttachedCommand : Command
+    {
+        private const string StreamPropertyExceptionMessage =
+            "This property cannot be used when attaching to already running process.";
+        private readonly Process process;
+        private readonly Task<CommandResult> commandResultTask;
+        
+        internal AttachedCommand(
+            Process process,
+            bool throwOnError,
+            TimeSpan timeout,
+            CancellationToken cancellationToken)
+        {
+            this.process = process;
+            var processMonitoringTask = System.Threading.Tasks.Task.Run(() => process.WaitForExit());
+            var processTask = ProcessHelper.CreateProcessTask(this.process, processMonitoringTask, throwOnError, timeout, cancellationToken);
+            this.commandResultTask = processTask.ContinueWith(continuedTask => new CommandResult(continuedTask.Result, this));
+        }
+
+        public override Process Process
+        {
+            get
+            {
+                this.ThrowIfDisposed();
+                return this.process;
+            }
+        }
+
+        public override IReadOnlyList<Process> Processes
+        {
+            get
+            {
+                this.ThrowIfDisposed();
+                return new ReadOnlyCollection<Process>(new[] { this.process });
```

Comment:
Good point. Instead of copying the solution from `ProcessCommand`, I have, however, used `Lazy<T>`, so that it is IMO easier to read and thread safe (although I'm not sure if that second property is desired here). What do you think about it? (I'll be pulling the changes soon).

Reply:
You mean that the mentioned property of the property needs only to be guaranteed when we're accessing it in a "raceless" way? Fair enough, I'd say, nevertheless it is given free, so why not have it.